### PR TITLE
Include more debug information in page transport error

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java [hunk 1]
```diff
@@ -535,7 +537,21 @@ public PagesResponse handle(Request request, Response response)
 
             // otherwise we must have gotten an OK response, everything else is considered fatal
             if (response.getStatusCode() != HttpStatus.OK.code()) {
-                throw new PageTransportErrorException(format("Expected response code to be 200, but was %s %s: %s", response.getStatusCode(), response.getStatusMessage(), request.getUri()));
+                StringBuilder body = new StringBuilder();
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(response.getInputStream()))) {
+                    // Get up to 1000 lines for debugging
+                    for (int i = 0; i < 1000; i++) {
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java]: I'd limit the overall size also.  You don't want 1000 1MB lines :)

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java]: Good point. Added a 100KB limit

**Comment by haozhun** [comment]
Looks good

File: presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java [hunk 2]
```diff
@@ -535,7 +537,22 @@ public PagesResponse handle(Request request, Response response)
...
+                        String line = reader.readLine();
+                        // Don't output more than 100KB
+                        if (line == null || body.length() + line.length() > 100 * 1024) {
+                            break;
+                        }
+                        body.append(line);
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java]: You probably want to append a line feed after you append `line`.

**Comment by haozhun** [comment]
I would copy the `try` block into a standalone java program, and feed it a file, to make sure the basic functionalities work. I will also use it to verify the 1000 line and 100KB limit.

File: presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java (hunk 2 above)
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java]: Oh ya, good catch!

**Comment by cberner** [comment]
Tested, and both worked fine

**Comment by haozhun** [comment]
I just realized it would be better if we made a utility method that takes an inputstream. This way, we can have real tests in the code. But since you've merged, never mind.

**Pull request merged**